application/utils.py

```
@@ -77,6 +86,7 @@ def validate_password(password):
         reason = f'Password should have at least one of the symbols [{"".join(special_symbols)}]'
 
     if reason is not None:
+        logger.error(f'Failed to validate password, reason: {reason}')
```

Ruff

Logging statement uses f-string

(G004)